src/main/render/shared/includes.ts

```diff
@@ -13,27 +13,64 @@ import { Resolver } from '../../resolver'
 import { DefinitionType, INamespacePath, IRenderState } from '../../types'
 import { COMMON_IDENTIFIERS } from './identifiers'
 
-function fieldTypeUsesThrift(fieldType: FieldType): boolean {
+function fieldTypeUsesThrift(
+    fieldType: FieldType,
+    state: IRenderState,
+): boolean {
     switch (fieldType.type) {
         case SyntaxType.I64Keyword:
             return true
 
         case SyntaxType.MapType:
             return (
-                fieldTypeUsesThrift(fieldType.keyType) ||
-                fieldTypeUsesThrift(fieldType.valueType)
+                fieldTypeUsesThrift(fieldType.keyType, state) ||
+                fieldTypeUsesThrift(fieldType.valueType, state)
             )
 
         case SyntaxType.ListType:
         case SyntaxType.SetType:
-            return fieldTypeUsesThrift(fieldType.valueType)
+            return fieldTypeUsesThrift(fieldType.valueType, state)
+        case SyntaxType.Identifier:
+            const {
+                definition,
+                namespace,
```

Comment:
To fix the namespacing issue we could add a check here.
If the definition's namespace is not part of the parent namespace then add it to the includes of the parent